### PR TITLE
Allow for cycling buildings and variants backwards

### PR DESCRIPTION
**Pull request by EmeraldBlock**
This makes cycling buildings or variants go backwards if the Rotate CCW Modifier key (default SHIFT) is pressed. This has the consequence of changing the meaning of the modifier key.

### CODE REVIEW

File: src/js/game/hud/parts/base_toolbar.js [hunk 1]
```diff
@@ -160,9 +160,16 @@ export class HUDBaseToolbar extends BaseHUDPart {
         }
 
         let newBuildingFound = false;
+        const direction = this.root.keyMapper.getBinding(KEYMAPPINGS.placement.rotateInverseModifier)
```
Comment (tobspr) [inline src/js/game/hud/parts/base_toolbar.js:163]: Linter issue

File: src/js/game/hud/parts/base_toolbar.js [hunk 2]
```diff
@@ -160,9 +160,16 @@ export class HUDBaseToolbar extends BaseHUDPart {
...
 
         let newBuildingFound = false;
+        const direction = this.root.keyMapper.getBinding(KEYMAPPINGS.placement.rotateInverseModifier)
+            .pressed
+            ? -1
+            : 1;
         let newIndex = this.lastSelectedIndex;
-        for (let i = 0; i < this.primaryBuildings.length; ++i, ++newIndex) {
+        for (let i = 0; i < this.primaryBuildings.length; ++i, newIndex += direction) {
             newIndex %= this.primaryBuildings.length;
```
Comment (tobspr) [inline src/js/game/hud/parts/base_toolbar.js:169]: `newIndex = safeModulo(newIndex + direction)` and remove the `++newIndex` from the for loop or ?

File: src/js/game/hud/parts/building_placer_logic.js [hunk 3]
```diff
@@ -467,7 +467,14 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
                 index = 0;
                 console.warn("Invalid variant selected:", this.currentVariant.get());
             }
-            const newIndex = (index + 1) % availableVariants.length;
+            const direction = this.root.keyMapper.getBinding(KEYMAPPINGS.placement.rotateInverseModifier)
+                .pressed
+                ? -1
+                : 1;
+            let newIndex = (index + direction) % availableVariants.length;
```
Comment (tobspr) [inline src/js/game/hud/parts/building_placer_logic.js:474]: see above.

Also don't use `rotateInverseModifier` -> Instead lets add a new bindign

**Review by tobspr — CHANGES_REQUESTED** [review]

File: src/js/game/hud/parts/building_placer_logic.js (hunk 3 above)
Comment (Vlad1slaw) [reply src/js/game/hud/parts/building_placer_logic.js:474]: Another key? Omg...
Same feature for Cycle buildings("Tab")?
So boring to cross half of screen, just to pick trash can...
Or ability to use scroll wheel with an extra binding(ofcource different from one used here :) )

Comment (EmeraldBlock) [reply src/js/game/hud/parts/building_placer_logic.js:474]: I personally feel that adding another binding would be excessive - I've always treated `(E ->) TAB -> T -> R` as a group of bindings that let you navigate the building options, and I think a single `backwardsModifier` or something would make the most sense

Comment (tobspr) [reply src/js/game/hud/parts/building_placer_logic.js:474]: Yeah I think we should unify it into one binding

Comment (EmeraldBlock) [reply src/js/game/hud/parts/building_placer_logic.js]: Should rotateInverseModifier be changed to something else, and/or should the current translations of it be reverted?

Comment (tobspr) [reply src/js/game/hud/parts/building_placer_logic.js]: I'd:
1) Get rid of rotateInverseModifier
2) Introduce a new binding "Backwards modifier"

**Comment by tobspr** [comment]
Need to have a look

**Comment by tobspr** [comment]
I redid this PR because there were some issues and it would have broken all upcoming PR's regarding translations

**Pull request closed without merging**